src/user/services/user-acl.service.ts

```diff
@@ -3,16 +3,21 @@ import { ROLE } from './../../auth/constants/role.constant';
 import { Action } from './../../shared/acl/action.constant';
 import { User } from '../entities/user.entity';
 import { BaseAclService } from '../../shared/acl/acl.service';
+import { Actor } from './../../shared/acl/actor.constant';
 
 @Injectable()
 export class UserAclService extends BaseAclService {
   constructor() {
     super();
+    // Admin can do all action
     this.canDo(ROLE.ADMIN, [Action.Manage]);
-    this.canDo(ROLE.USER, [Action.Read, Action.Update], this.isResourceOwner);
+    //user can read himself or any other user
+    this.canDo(ROLE.USER, [Action.Read]);
+    // user can only update himself
+    this.canDo(ROLE.USER, [Action.Update], this.isUserHimself);
   }
 
-  isResourceOwner(subject: User, user: User): boolean {
-    return subject.id === user.id;
+  isUserHimself(resource: User, actor: User): boolean {
```

Comment:
```suggestion
  isUserHimself(resource: User, actor: Actor): boolean {
```
Need to confirm if `Actor` type here is better?